src/bst.py

```diff
@@ -0,0 +1,231 @@
+"""Implementation of a Binary Search Tree."""
+
+from my_queue import Queue
+from stack import Stack
+
+
+class Node(object):
+    """Node class."""
+
+    def __init__(self, val):
+        """Node instantiation."""
+        self.val = val
+        self.leftChild = None
+        self.rightChild = None
+
+
+class BinarySearchTree(object):
+    """Binary Search Tree Class."""
+
+    def __init__(self):
+        """Binary Search Tree instantiation."""
+        self.root = None
+        self._size = 0
+
+    def insert(self, val):
+        """Insert value into Binary Search Tree."""
+        if not self.contains(val):
```

Comment:
There is probably better way to do this so you don't unnecessarily increase your time complexity.